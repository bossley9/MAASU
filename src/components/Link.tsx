import React, { FC, ReactNode } from 'react'
import { Link as ReachLink } from '@reach/router'

type LinkProps = {
  children: ReactNode
  to: string
  className?: string
}

const newTabProps = { target: '_blank', rel: 'noreferrer noopener' }
const classNames = 'c-text-main tdu hov-c-primary-main'

export const Link: FC<LinkProps> = ({ children, to, className = '' }) => (
  <ReachLink to={to} className={`${classNames} ${className}`}>
    {children}
  </ReachLink>
)

type AProps = {
  children: ReactNode
  href: string
  className?: string
}

export const A: FC<AProps> = ({ children, href, className = '' }) => (
  <a {...newTabProps} href={href} className={`${classNames} ${className}`}>
    {children}
  </a>
)
