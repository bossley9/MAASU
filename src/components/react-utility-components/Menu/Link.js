import React from 'react'
import PropTypes from 'prop-types'
// import modules from './Menu.module.scss'

class Link extends React.Component {
  render() {
    const {
      children,
      className,
      dropdown,
      horizontal,
      href,
      isSubMenuTrigger,
      linkWrapper,
      ...props
    } = this.props

    const classes = typeof className === 'string' ? className.split(' ') : []
    // classes.push(modules.link)

    if (horizontal) {
      classes.push('horizontal-ruc')
    }

    // no href exists
    let link = (
      <span className={classes.join(' ')} {...props}>
        {children}
      </span>
    )

    if (dropdown) {
      link = (
        <button className={classes.join(' ')} {...props}>
          {children}
        </button>
      )
    }

    if (href) {
      // linkWrapper
      link = linkWrapper(href, children)

      // default link
      if (!link) {
        link = (
          <a className={classes.join(' ')} href={href}>
            {children}
          </a>
        )
      }
    }

    // is not a trigger
    if (!isSubMenuTrigger) {
      link = (
        <li className={classes.join(' ')} {...props}>
          {link}
        </li>
      )
    }

    return link
  }
}

Link.propTypes = {
  children: PropTypes.node.isRequired,
  dropdown: PropTypes.bool,
  horizontal: PropTypes.bool,
  href: PropTypes.string,
  isSubMenuTrigger: PropTypes.bool,
  linkWrapper: PropTypes.func,
}

Link.defaultProps = {
  dropdown: false,
  horizontal: false,
  href: null,
  isSubMenuTrigger: false,
  linkWrapper: () => {},
}

export default Link
