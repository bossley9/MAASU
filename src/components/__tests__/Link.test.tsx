import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { A, Link } from '../Link'

describe('A', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<A href="https://maasu.org">test</A>, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('renders a correct snapshot', () => {
    const test = renderer.create(<A href="https://maasu.org">test text</A>)
    expect(test.toJSON()).toMatchSnapshot()
  })
})

describe('Link', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Link to="/home">test</Link>, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('renders a correct snapshot', () => {
    const test = renderer.create(<Link to="/contact">test text</Link>)
    expect(test.toJSON()).toMatchSnapshot()
  })
})
