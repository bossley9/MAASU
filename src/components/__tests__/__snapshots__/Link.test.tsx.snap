// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A renders a correct snapshot 1`] = `
<a
  className="c-text-main tdu hov-c-primary-main "
  href="https://maasu.org"
  rel="noreferrer noopener"
  target="_blank"
>
  test text
</a>
`;

exports[`Link renders a correct snapshot 1`] = `
<a
  className="c-text-main tdu hov-c-primary-main "
  href="/contact"
  onClick={[Function]}
>
  test text
</a>
`;
